src/dex-ui-components/presets/SwapTokensForm/SwapTokensForm.tsx

```diff
@@ -207,64 +235,49 @@ for ${formValues.tokenToReceive.amount.toFixed(8)} ${formValues.tokenToReceive.s
-      displayAmount: tokenToReceiveAmount.gt(0) ? String(tokenToReceiveAmount) : "",
-    };
-    updateTokenToReceiveAmounts(updatedTokenToReceive.amount, updatedTokenToReceive.displayAmount);
+    const tokenToTradeQty = updatedToken.amount;
+    if (tokenToTradeQty) {
+      fetchSwapPair(
+        updatedToken.tokenMeta.tokenId ?? "",
+        formValues.tokenToReceive.tokenMeta.tokenId ?? "",
+        updatedToken.amount === 0 ? 1 : updatedToken.amount
+      );
+    } else {
+      swapTokensForm.setValue("tokenToReceive.amount", 0);
+      swapTokensForm.setValue("tokenToReceive.displayAmount", "");
+    }
   }
 
   function handleTokenToTradeSymbolChanged(updatedToken: TokenState) {
     swapTokensForm.setValue("tokenToReceive", InitialSwapFormState.tokenToReceive);
   }
 
   function handleTokenToReceiveSymbolChanged(updatedToken: TokenState) {
-    const { tokenToTradeData, tokenToReceiveData } = getPairedTokenData(
+    fetchSwapPair(
       formValues.tokenToTrade.tokenMeta.tokenId ?? "",
       updatedToken.tokenMeta.tokenId ?? "",
-      props.tokenPairs ?? []
-    );
-    const tokenToTradeBalance = getTokenBalance(tokenToTradeData?.tokenMeta.tokenId ?? "", props.pairedAccountBalance);
-    const updatedTokenToTrade = {
-      ...formValues.tokenToTrade,
-      tokenMeta: tokenToTradeData?.tokenMeta ?? formValues.tokenToTrade.tokenMeta,
-      symbol: tokenToTradeData?.symbol ?? formValues.tokenToTrade.symbol,
-      balance: tokenToTradeBalance,
-    };
-    swapTokensForm.setValue("tokenToTrade.symbol", updatedTokenToTrade.symbol);
-    swapTokensForm.setValue("tokenToTrade.balance", updatedTokenToTrade.balance);
-    swapTokensForm.setValue("tokenToTrade.tokenMeta", updatedTokenToTrade.tokenMeta);
-    const tokenToReceiveBalance = getTokenBalance(
-      tokenToReceiveData?.tokenMeta.tokenId ?? "",
-      props.pairedAccountBalance
+      formValues.tokenToTrade.amount === 0 ? 1 : formValues.tokenToTrade.amount
     );
-    const updatedTokenToReceive = {
-      ...formValues.tokenToReceive,
-      symbol: tokenToReceiveData?.symbol,
-      tokenMeta: tokenToReceiveData?.tokenMeta ?? DefaultTokenMeta,
-      balance: tokenToReceiveBalance,
-    };
-    swapTokensForm.setValue("tokenToReceive.symbol", updatedTokenToReceive.symbol);
-    swapTokensForm.setValue("tokenToReceive.balance", updatedTokenToReceive.balance);
-    swapTokensForm.setValue("tokenToReceive.tokenMeta", updatedTokenToReceive.tokenMeta);
-    fetchPoolLiquidity(updatedTokenToTrade, updatedTokenToReceive);
-    props.fetchPairInfo(updatedTokenToReceive.tokenMeta.pairAccountId ?? "");
+
+    /* TODO: This is done as a design approach with SC team meanwhile 
+    SC team is trying merge fetchPairInfo and SwapPairFetch into single one. */
+
+    if (formValues.tokenToTrade.amount === 0) {
+      swapTokensForm.setValue("tokenToTrade.amount", 1);
+      swapTokensForm.setValue("tokenToTrade.displayAmount", "1");
+    }
```

Comment:
So due to above stated reason, SC team and I decided to update the Input token with 1 amount when user selected token to receive symbol if its 0.